### PR TITLE
Create workspace only when needed

### PR DESCRIPTION
**Pull request by Katka92**
Workspace is reused when next test needs the same project and param removeAfterTest is false.

### CODE REVIEW

**Comment by Katka92** [comment]
@rhopp can you review please?

**Comment by rhopp** [comment]
[test]

**Comment by Katka92** [comment]
@rhopp changes merged

File: arquillian-extension-che/src/main/java/com/redhat/arquillian/che/annotations/Workspace.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.redhat.arquillian.che.annotations;
```
Comment (rhopp) [inline arquillian-extension-che/src/main/java/com/redhat/arquillian/che/annotations/Workspace.java]: Add license, please ;-)

File: arquillian-extension-che/src/main/java/com/redhat/arquillian/che/CheWorkspaceManager.java [hunk 2]
```diff
@@ -43,71 +58,90 @@
...
+            CheWorkspaceService.deleteWorkspace(cheWorkspaceInstanceProducer.get(), bearerToken);
+            //the attribute deleted is used, because workspace can't be set at null - aruqillian Exception
+            cheWorkspaceInstanceProducer.get().setDeleted(true);
+        }
+    }
 
-        return workspace;
+    private void createWorkspace(Workspace workspaceAnnotation) {
+        CheWorkspaceProvider provider = cheWorkspaceProviderInstanceProducer.get();
+        String path;
```
Comment (rhopp) [inline arquillian-extension-che/src/main/java/com/redhat/arquillian/che/CheWorkspaceManager.java]: What is this variable used for?

**Pull request merged**